lib/apytypes/_array_functions.py

```diff
@@ -2167,6 +2169,281 @@ def fullrange(
         raise ValueError("Could not determine array type")
 
 
+@overload
+def export_csv(
+    a: APyFixedArray,
+    fname: str,
+    delimiter: str = ",",
+    layout: str = "default",
+) -> None: ...
+
+
+@overload
+def export_csv(
+    a: APyFloatArray,
+    fname: str,
+    delimiter: str = ",",
+    layout: str = "default",
+) -> None: ...
+
+
+def export_csv(
+    a: APyArray,
+    fname: str,
+    delimiter: str = ",",
+    layout: str = "default",
+) -> None:
+    """
+    Store the bit values of an array to a CSV file.
+
+    The function supports multi-dimensional arrays:
+    - For 1-D arrays, all elements are written on a single line.
+    - For 2-D arrays, each row of the array is written on a separate line.
+    - Higher-dimensional arrays must be handled manually unless supported by the chosen layout.
+
+    By default, a 1-D array will be exported as a single row CSV, and a 2-D array will be exported as multiple rows.
+
+    .. attention::
+        The index order of arrays may differ between APyTypes and other programs.
+
```

Comment:
```suggestion
.. hint::
    To export a complex-valued array, use separate CSV files for real and imaginary parts.
```
or something similar.